src/vs/workbench/contrib/accessibility/browser/accessibleView.ts

```diff
@@ -0,0 +1,169 @@
+import { CodeEditorWidget, ICodeEditorWidgetOptions } from 'vs/editor/browser/widget/codeEditorWidget';
+import { ITextModel } from 'vs/editor/common/model';
+import { IModelService } from 'vs/editor/common/services/model';
+import { LinkDetector } from 'vs/editor/contrib/links/browser/links';
+import { localize } from 'vs/nls';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IContextViewDelegate, IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { IInstantiationService, createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { SelectionClipboardContributionID } from 'vs/workbench/contrib/codeEditor/browser/selectionClipboard';
+import { getSimpleEditorOptions } from 'vs/workbench/contrib/codeEditor/browser/simpleEditorOptions';
+import { IDisposable } from 'xterm';
+
+export interface IAccessibleContentProvider { id: string; provideContent(): string; onClose(): void; options: IAccessibleViewOptions }
+export const IAccessibleViewService = createDecorator<IAccessibleViewService>('accessibleViewService');
+
+export interface IAccessibleViewService {
+	readonly _serviceBrand: undefined;
+	show(providerId: string): void;
+	registerProvider(provider: IAccessibleContentProvider): IDisposable;
+}
+
+export interface IAccessibleViewOptions {
+	ariaLabel: string;
+}
+
+class AccessibleView extends Disposable {
+	private _editorWidget: CodeEditorWidget;
+	private _editorContainer: HTMLElement;
+
+	constructor(
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@IModelService private readonly _modelService: IModelService,
+		@IContextViewService private readonly _contextViewService: IContextViewService
+	) {
+		super();
+		this._editorContainer = document.createElement('div');
+		this._editorContainer.classList.add('accessible-view');
+		const codeEditorWidgetOptions: ICodeEditorWidgetOptions = {
+			contributions: EditorExtensionsRegistry.getSomeEditorContributions([LinkDetector.ID, SelectionClipboardContributionID, 'editor.contrib.selectionAnchorController'])
+		};
+		const editorOptions: IEditorConstructionOptions = {
+			...getSimpleEditorOptions(),
+			lineDecorationsWidth: 6,
+			dragAndDrop: true,
+			cursorWidth: 1,
+			wrappingStrategy: 'advanced',
+			wrappingIndent: 'none',
+			padding: { top: 2, bottom: 2 },
+			quickSuggestions: false,
+			renderWhitespace: 'none',
+			dropIntoEditor: { enabled: true },
+			accessibilitySupport: this._configurationService.getValue<'auto' | 'off' | 'on'>('editor.accessibilitySupport'),
+			cursorBlinking: this._configurationService.getValue('terminal.integrated.cursorBlinking'),
+			readOnly: true
+		};
+		this._editorWidget = this._register(this._instantiationService.createInstance(CodeEditorWidget, this._editorContainer, editorOptions, codeEditorWidgetOptions));
```

Comment:
The comment about lazy loading was resolved but it wasn't done, might have been an accident?

Reply:
I think I addressed that as this object only gets created when `show` is called

Reply:
https://github.com/microsoft/vscode/pull/184435/files/47264ae67cd6155a4498ddcd817da7307adc4d7e#diff-57bdd64acf33af25dee4c88c656bf7548ab00a0d9eaebaae5e627ddb299a8516R160

Reply:
Yep, my bad